quesma/ingest/processor.go

```diff
@@ -560,14 +564,30 @@ func (ip *IngestProcessor) processInsertQuery(ctx context.Context,
 		}
 		processed = append(processed, result)
 	}
+	jsonData = processed
+
+	encodings := make(map[schema.FieldEncodingKey]string)
+	// we are doing two passes, e.g. calling transformFieldName twice
+	// first time we populate encodings map
+	// second time we do field encoding
+	// This is because we need to know all field encodings before we start
+	// transforming fields, otherwise we would mutate json and populating encodings
+	// which would introduce side effects
+	for _, jsonValue := range jsonData {
+		transformFieldName(jsonValue, func(field string) string {
+			return populateFieldEncodings(field, tableName, encodings)
+		})
+	}
+	if ip.schemaRegistry != nil {
+		ip.schemaRegistry.UpdateFieldEncodings(encodings)
+	}
 	// Do field encoding here, once for all jsons
 	// This is in-place operation
 	for _, jsonValue := range jsonData {
 		transformFieldName(jsonValue, func(field string) string {
-			return fieldToColumnEncoder(field)
+			return util.FieldToColumnEncoder(field)
```

Comment:
Now when I think about those usages of `FieldToColumnEncoder`, perhaps this function should really part of schema registry? I could imagine registry being an interface or perhaps having it parametrized. But of course today it's not an option given he'd still use that to assemble the response :)

Reply:
Maybe, for it's just utility function, but it could be moved or invoked from `schema.Registry`